raiden/api/rest.py

```diff
@@ -94,91 +107,154 @@ class RestAPI(object):
-                return self.channel_schema.dumps(api_result)
+    def get_channel(self, channel_address):
+        channel = self.raiden_api.get_channel(channel_address)
+        return self.channel_schema.dumps(channel)
+
+    def get_partners_by_token(self, token_address):
+        return_list = []
+        raiden_service_result = self.raiden_api.get_channel_list(token_address)
+        for result in raiden_service_result:
+            return_list.append({
+                'partner_address': result.partner_address,
+                'channel': url_for(
+                    # TODO: Somehow nicely parameterize this for future versions
+                    'v1_resources.channelsresourcebychanneladdress',
+                    channel_address=result.channel_address
+                ),
+            })
+
+        schema_list = PartnersPerTokenList(return_list)
+        result = self.partner_per_token_list_schema.dumps(schema_list)
+        return result
 
-            if status == 'open':
-                raise Exception('nothing to do here')
-        raise Exception()
+    def patch_channel(self, channel_address, balance=None, state=None):
+        # find the channel
+        channel = self.raiden_api.get_channel(channel_address)
+        # if we patch with `balance` it's a deposit
+        if balance is not None and state is None:
+            raiden_service_result = self.raiden_api.deposit(
+                channel.token_address,
+                channel.partner_address,
+                balance
+            )
+            return self.channel_schema.dumps(raiden_service_result)
+
+        elif state is not None and balance is None:
+            current_state = channel.state
+            if state == 'closed':
+                if current_state != 'open':
+                    return make_response(
+                        httplib.CONFLICT,
+                        'Attempted to close an already closed channel'
+                    )
+                raiden_service_result = self.raiden_api.close(
+                    channel.token_address,
+                    channel.partner_address
+                )
+                return self.channel_schema.dumps(raiden_service_result)
+            elif state == 'settled':
+                if current_state == 'settled' or current_state == 'open':
+                    return make_response(
+                        'Attempted to settle a channel at its {} state'.format(current_state),
+                        httplib.CONFLICT,
+                    )
+                raiden_service_result = self.raiden_api.settle(
+                    channel.token_address,
+                    channel.partner_address
+                )
+                return self.channel_schema.dumps(raiden_service_result)
+            else:
+                return make_response(
+                    'Provided invalid channel state {}'.format(state),
+                    httplib.BAD_REQUEST,
+                )
+
+        return make_response(
```

Comment:
Bike shedding: I believe this return would be better placed at the top of the function with a `state is None and balance is None` guard, that way the other checks may be simplified.